server/controllers/command.js

```diff
@@ -722,39 +724,70 @@ const performAction = async (req, res) => {
     }
 
     if (
-      await isBotPresent(
+      !(await isBotPresent(
         payload.team.id,
         payload.channel.id,
         payload.user.id,
         payload.response_url
-      )
+      ))
     ) {
-      const [isAssociated, user] = await auth.checkSlackAssociationStatus(
-        payload.user.id
+      return;
+    }
+    // Assuming that all actions require user to be data.world authenticated
+    const [isAssociated, user] = await auth.checkSlackAssociationStatus(
+      payload.user.id
+    );
+    if (!isAssociated) {
+      // User is not associated begin association process.
+      await beginSlackAssociation(
+        payload.user.id,
+        payload.channel.id,
+        payload.team.id
       );
-      if (isAssociated) {
-        // subscribe or unsubscribe to/from resource.
-        collection.forEach(payload.actions, async action => {
-          switch (action.type) {
-            case "button":
-              await handleButtonAction(payload, action, user);
-              break;
-            case "select":
-              await handleMenuAction(payload, action, user);
-              break;
-            default:
-              console.warn("Unknown action type : ", action.type);
-              break;
-          }
-        });
-      } else {
-        // User is not associated begin association process.
-        await beginSlackAssociation(
-          payload.user.id,
-          payload.channel.id,
-          payload.team.id
-        );
-      }
+      return;
+    }
+
+    if ("callback_id" in payload) {
+      // Handle legacy Slack actions
+      // https://api.slack.com/messaging/attachments-to-blocks#callback_id_replacement
+
+      // subscribe or unsubscribe to/from resource.
+      collection.forEach(payload.actions, async action => {
+        switch (action.type) {
+          case "button":
+            await handleButtonAction(payload, action, user);
+            break;
+          case "select":
+            await handleMenuAction(payload, action, user);
+            break;
+          default:
+            console.warn("Unknown action type : ", action.type);
+            break;
+        }
+      });
+    } else {
+      // Handle new Slack block kit actions
+      collection.forEach(payload.actions, async action => {
+        const actionid = action.action_id
+        if (Object.values(AUTHORIZATION_ACTIONS).includes(actionid)) {
+          const { requestid, agentid, datasetid } = JSON.parse(action.value);
+          await handleDatasetRequestAction({
+            channelid: payload.channel.id,
+            userid: payload.user.id,
+            triggerid: payload.trigger_id,
+            responseUrl: payload.response_url,
+            message: payload.message,
```

Comment:
@katz12 I overlooked that the original message is actually passed in the action payload we receive. So we don't have to add the extra `history` scopes to update the original message.

In order to fully support threading later on, we may need to add the `history` scopes.

Reply:
Oh nice!

Why would we need the scope for threading then? Or are you just unsure whether we have everything we need in this message?

Reply:
Not specifically for threading, per se. It would be useful for handling multiple notifications about the same request.

Let's say a user requests access to a dataset and the request is accepted on data.world (not through Slack). Slack would receive an `authorization_request.created` message followed by an `authorization_request.accepted` message some time later. Since the request was not accepted on Slack, the original `authorization_request.created` Slack message would not update. Users could still try to accept or approve the request on Slack, but this would show error messages, as the request is now invalid.

Ideally, we would update the original message `authorization_request.created` after we receive an `authorization_request.accepted` message for the same request. But we don't have access to the `authorization_request.created` message this time, so we would need to store the request data to build the original message again, or use the `history` scope to fetch the original message.

Reply:
> Let's say a user requests access to a dataset and the request is accepted on data.world (not through Slack).

Yeah I was thinking about this case. This makes sense for future functionality, and I see that it is different than a message being acted upon because there is no reference to the slack message in the raw d.w webhook.

Reply:
Yep, this is a scenario where we would likely need to create a requests table to store the id of the parent message.